tcms/core/tests/test_middleware.py

```diff
@@ -16,3 +23,23 @@ def test_unapplied_migrations(self):
             'container">documentation</a>'
         response = self.client.get('/', follow=True)
         self.assertContains(response, unapplied_migration_message)
+
+
+class TestCsrfDisableMiddleware(unittest.TestCase):
+    def test_prcess_view(self):
+        request = HttpRequest()
+        middleware = CsrfDisableMiddleware()
+        middleware.process_view(request,
+                                _callback=None,
+                                _callback_args=None,
+                                _callback_kwargs=None)
+        self.assertEqual(request._dont_enforce_csrf_checks, True)
+
+
+class TestCheckSettingsMiddleware(test.TransactionTestCase):
+    @mock.patch('django.contrib.messages.add_message')
+    def test_process_request(self, mocked_add_message):
+        request = HttpRequest()
+        middleware = CheckSettingsMiddleware()
+        middleware.process_request(request)
+        mocked_add_message.assert_called_once()
```

Comment:
This test scenario is no good. It doesn't setup conditions in the SUT and doesn't assert any of the desired results other than the fact that the middleware was executed. This tells us nothing about whether or not this middleware executed successfully.  See the rest of this file for example.

Reply:
I have updated this test according to `TestCheckUnappliedMigrationsMiddleware`